services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/StepIntersection.java

```diff
@@ -92,6 +93,81 @@ public Point location() {
   @Nullable
   public abstract List<Boolean> entry();
 
+  /**
+   * A list representing the "form of way" values for all roads at the step intersection.
+   * This list has a 1:1 correspondence with the {@link #bearings()} and {@link #entry()} lists.
+   * Each element is one of the following:
+   * - freeway,
+   * - multiple_carriageway,
+   * - single_carriageway,
+   * - roundabout,
+   * - ramp,
+   * - service_road,
+   * - pedestrian_zone,
+   * - unknown
+   * Example:
+   * "form_of_way": [
+   * "ramp",
+   * "single_carriageway",
+   * "freeway"
+   * ]
+   *
+   * @return A list of form of way values for all roads at the step intersection
+   */
+  @Nullable
+  @SerializedName("form_of_way")
+  public abstract List<String> formOfWay();
+
+  /**
+   * A list representing the geometry of each road at the step intersection.
+   * This list has a 1:1 correspondence with the {@link #bearings()} and {@link #entry()} lists.
+   * Each value represents the geometry of a segment up to the next intersection (a SIRN segment).
```

Comment:
Let's not expose a SIRN term here, they are for internal use only. The geometries here also not strictly tied to SIRNs, it's just the most convenient way right now to find a segment till next intersection.

```suggestion
   * Each value represents the geometry of a segment up to the next intersection.
```